packages/gatsby-image/README.md

```diff
@@ -94,6 +94,34 @@ module.exports = {
 }
 ```
 
+### Dynamically including polyfills
+
+You can use `gatsby-image` without adding it to `gatsby-config.js` (it Just Works™). However, lazy loading (e.g. blur-up) may not work in browsers lacking support for `IntersectionObserver` (e.g. Safari and IE) and `object-fit`/`object-position` may not work in browsers lacking support for these CSS properties (e.g. IE). 
+
+To support these features in all browsers, add `gatsby-image` to `gatsby-config.js` and set the `loadPolyfills` option to true. This tells `gatsby-image` to dynamically import the [intersection-observer](https://github.com/w3c/IntersectionObserver#readme) and [object-fit-images](https://github.com/bfred-it/object-fit-images) polyfills in browsers that need them. (These polyfills will not be loaded in browsers that already support these features.)
```

Comment:
What do you think of _just_ loading object-fit-images here? It seems like intersection-observer is more "app level" polyfill whereas object-fit-images is actually needed and makes a ton of sense for an image component to provide opt-in support for that polyfill--additionally because this component _needs_ something extra to make the polyfill useful.

Primarily I mention this because gatsby-link also uses an intersection observer, and we sorta treat that as "progressively enhanced," i.e. pages are prefetched in certain conditions _and_ if the API is available. I think doing something similar here makes sense, but I'm open to your opinion here!

Reply:
Sounds good!

I agree: object-fit/position really *only* affects images, so it makes sense to tie that polyfill to `gatsby-image` (since it's the only component likely to use it). It also affects the alignment of the image, which is crucial in some cases. 

On the other hand, IO is app-level and only affects the `gatsby-image` loading effect, which I agree is a nice-to-have.

Should we explain how to dynamically polyfill IO via `gatsby-browser` in this part of the README? I'm torn since I don't want the preamble to get too long, but I also remember how long it took me to notice, research and solve the `gatsby-image` polyfilling issues, so it would be nice to spare future devs that work.

Reply:
Those are are all really good points! I think what makes the most sense is a doc on polyfilling + gatsby, i.e. how to dynamically load polyfills based on certain conditions. It could even cover using something like [polyfill.io](https://polyfill.io/v2/docs/).

So... short answer: I think we take out the polyfill for now, and then link to a stub, perhaps? I like the idea of linking to more general purpose, reusable documentation rather than re-inventing the wheel as far as documenting things goes!